src/snapred/backend/data/GroceryService.py

```diff
@@ -814,8 +814,14 @@ def fetchGroceryList(self, groceryList: List[GroceryListItem]) -> List[Workspace
                             loader="ReheatLeftovers",
                         )
                 case "diffcal_table":
+                    if not item.version:
+                        item.version = self.dataService._getVersionFromCalibrationIndex(
+                            item.runNumber, item.useLiteMode
+                        )
+                    record = self.dataService.readCalibrationRecord(item.runNumber, item.useLiteMode, item.version)
+                    item.runNumber = record.runNumber
```

Comment:
this search was giving Malcolm some issues

Reply:
I guess this may have issue if they try to use the default calib, as that has no record, but then what runnumber would be used if it referred to default?  this would need a branch to cover that, or to point to some code that already handles it.